### PR TITLE
Events data

### PR DESCRIPTION
**Pull request by Kvtran3546**

### CODE REVIEW

File: server/data/users.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ const createUser = async (
   gender = validation.checkGender(gender);
 
   const user_collection = await users();
-  hashed_password = await bcrypt.hash(password, 10);
+  hashed_password = await bcrypt.hash(password, 16);
```
Comment (manisaiprasad) [inline server/data/users.js]: Why this is chnaged

File: server/data/users.js [hunk 2]
```diff
@@ -32,7 +32,6 @@ const createUser = async (
     phone: phone,
     hashed_password: hashed_password,
     gender: gender,
```
Comment (manisaiprasad) [inline server/data/users.js:38]: Why is  this removed

File: server/data/users.js [hunk 3]
```diff
@@ -67,75 +67,9 @@ const getUserByEmail = async (email) => {
   return user;
 };
```
Comment (manisaiprasad) [inline server/data/users.js]: Why this is removed if this from merge conflicts make sure all changes from others are included

**Review by manisaiprasad — CHANGES_REQUESTED** [review]

**Review by SumanMohanty20012987 — APPROVED** [review]
Had to make the following changes
Changes ->
	data/event.js:
		1)line 92->In getAllEvents
					return events -> return events_list
		2)line 103-> In getEventsByTitle:
					added ``
		3)line 106->const event = await eventCollection.findOne({ event_title: title }); changed key
						
						
	
	
	validation.js:
		1)line 32: throw `Title must only contain letters or numbers`
		2)line 34: if (input.length < 4) throw `${name} should be of length 4 or greater`;

File: server/data/users.js [hunk 4]
```diff
@@ -58,6 +61,23 @@ const getUserById = async (id) => {
...
+};
+
+const getEventById_Object = async (event_id) => {
+  event_id = validation.checkObjectId(event_id);
+  const eventCollection = await events();
+  const event = await eventCollection.findOne({ _id: ObjectId(event_id) });
+  if (event === null) throw new Error("No event with that id");
+  return event;
+};
+
```
Comment (manisaiprasad) [inline server/data/users.js]:
```suggestion
```

File: server/data/users.js [hunk 5]
```diff
@@ -110,6 +130,18 @@ const updateUser = async (
...
+  const users_list = await userCollection.find({}).toArray();
+  if (!users_list) {
+    throw new Error("Could not get all events.");
+  }
+  for (const element of users_list) {
+    element._id = element._id.toString();
+  }
+  return users_list; //changed from events to event_list
+};
+
```
Comment (manisaiprasad) [inline server/data/users.js]:
```suggestion
```

File: server/data/users.js [hunk 6]
```diff
@@ -192,6 +224,9 @@ const updateImageURL = async (userId, imageURL) => {
     { _id: ObjectId(userId) },
     { $set: { profile_photo_url: imageURL } }
   );
+  if (updatedInfo.modifiedCount === 0) {
+    throw 'could not update user successfully';
+  }
```
Comment (manisaiprasad) [inline server/data/users.js]:
```suggestion
```

File: server/data/users.js (hunk 4 above)
Comment (manisaiprasad) [reply server/data/users.js]: duplicated getUserById_Object

**Review by manisaiprasad — APPROVED** [review]

**Pull request merged**